--- 
title: "Tutorial for R MicrobiomeAnalysis package"
author: "Hua Zou"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [assets/book.bib, assets/packages.bib]
biblio-style: apalike
link-citations: yes
always_allow_html: true
description: "MicrobiomeAnalysis: A comprehensive R package for managing and analyzing microbiome data"
---

# Introduction

This tutorial describes how to perform bioinformatics data analysis of microbiota using metagenomic data. It will focus on data processing, diversity analysis, biomarker identification and association investigation from microbiota expression profile which derived from **amplicon (16s rRNA)** or **Whole genome (metagenomics)** sequencing data.

```{r, echo=FALSE, dpi=600, fig.align="center", out.width="100%", out.height="100%", fig.cap="Flowchat of MicrobiomeAnalysis"}
knitr::include_graphics("./figures/Schematic.png")
```



The data pre-processing module includes six main procedures:

+ **Data Transformation**
+ **Data Imputation**
+ **Data Normalization**
+ **Data Scaling**
+ **Data Trimming**
+ **Data Filtering**


The downstream analysis includes several modules:

+ **Diversity analysis**
+ **Ordination analysis**
+ **Clustering analysis**
+ **Differential analysis**
+ **Association analysis**


## Input data 

when using data pre-processing functions, it requires that the inputs which are from 16s or metagenomics data should be converted into phyloseq-class object:

+ **Types of data**. Users can handle 16s or metagenomics data coming from any platform on a transformed phyloseq-class object when they use `import_dada2` or `import_qiime2`.

+ **Transformation**. The package provides multiple methods to transform individual values, such as log10, 

+ **Imputation**. Missing value imputation methods also were provided by this package.

+ **Normalisation**. `MicrobiomeAnalysis` also has so many methods to normalize data sample by sample, which could remove technical effects. 

+ **Prefiltering**. Removing low abundant taxa also could be done by using `MicrobiomeAnalysis`.

+ **Data format**. Most functions of `MicrobiomeAnalysis` are using phyloseq-class object as input. Therefore, users could convert their inputs into phyloseq-class object.


## Methods


### Some background knowledge

We list here the main methodological or theoretical concepts you need to know to be able to efficiently apply `MicrobiomeAnalysis`:

+ **phyloseq-class, EpressionSet-class**: the former object contains otu table, sample data and taxa table etc which are usually used in metagenomic data analysis. The latter one comprises expression data and metadata of gene, metabolites etc.


### Key publications

+ **phyloseq object**. [@mcmurdie2013phyloseq] phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data.

+ **Permutational Multivariate Analysis of Variance (PERMANOVA)**. [@anderson2014permutational] Permutational multivariate analysis of variance (PERMANOVA).

+ **ALDEx2**. [@fernandes2014unifying] Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis.



## Outline of this tutorial

+ **Chapter [2]()** How to install `MicrobiomeAnalysis`

+ **Chapter [3]()** Data Processing


## Citation

Kindly cite by using citation("MicrobiomeAnalysis") if you think `MicrobiomeAnalysis` helps you. Alternative way is Zou H (2022). MicrobiomeAnalysis: An R package for analysis and visualization in metagenomics. R package version 1.0.3, <URL:https://github.com/HuaZou/MicrobiomeAnalysis/>.

