<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Tutorial for R MicrobiomeAnalysis package</title>
  <meta name="description" content="MicrobiomeAnalysis: A comprehensive R package for managing and analyzing microbiome data" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Tutorial for R MicrobiomeAnalysis package" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="MicrobiomeAnalysis: A comprehensive R package for managing and analyzing microbiome data" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Tutorial for R MicrobiomeAnalysis package" />
  
  <meta name="twitter:description" content="MicrobiomeAnalysis: A comprehensive R package for managing and analyzing microbiome data" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2023-01-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="how-to-install-microbiomeanalysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tutorial for R MicrobiomeAnalysis package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#input-data"><i class="fa fa-check"></i><b>1.1</b> Input data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#methods"><i class="fa fa-check"></i><b>1.2</b> Methods</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#some-background-knowledge"><i class="fa fa-check"></i><b>1.2.1</b> Some background knowledge</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#key-publications"><i class="fa fa-check"></i><b>1.2.2</b> Key publications</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#outline-of-this-tutorial"><i class="fa fa-check"></i><b>1.3</b> Outline of this tutorial</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.4</b> Citation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-to-install-microbiomeanalysis.html"><a href="how-to-install-microbiomeanalysis.html"><i class="fa fa-check"></i><b>2</b> How to install MicrobiomeAnalysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-to-install-microbiomeanalysis.html"><a href="how-to-install-microbiomeanalysis.html#installation"><i class="fa fa-check"></i><b>2.1</b> Installation</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-install-microbiomeanalysis.html"><a href="how-to-install-microbiomeanalysis.html#vignette"><i class="fa fa-check"></i><b>2.2</b> Vignette</a></li>
<li class="chapter" data-level="2.3" data-path="how-to-install-microbiomeanalysis.html"><a href="how-to-install-microbiomeanalysis.html#authors"><i class="fa fa-check"></i><b>2.3</b> Authors</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>3</b> Data Processing</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-processing.html"><a href="data-processing.html#loading-packages"><i class="fa fa-check"></i><b>3.1</b> Loading Packages</a></li>
<li class="chapter" data-level="3.2" data-path="data-processing.html"><a href="data-processing.html#importing-data"><i class="fa fa-check"></i><b>3.2</b> Importing Data</a></li>
<li class="chapter" data-level="3.3" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-levels"><i class="fa fa-check"></i><b>3.3</b> Extracting specific levels</a></li>
<li class="chapter" data-level="3.4" data-path="data-processing.html"><a href="data-processing.html#summarizing-specific-levels"><i class="fa fa-check"></i><b>3.4</b> Summarizing specific levels</a></li>
<li class="chapter" data-level="3.5" data-path="data-processing.html"><a href="data-processing.html#data-transformation"><i class="fa fa-check"></i><b>3.5</b> Data Transformation</a></li>
<li class="chapter" data-level="3.6" data-path="data-processing.html"><a href="data-processing.html#data-imputation"><i class="fa fa-check"></i><b>3.6</b> Data Imputation</a></li>
<li class="chapter" data-level="3.7" data-path="data-processing.html"><a href="data-processing.html#data-normalization"><i class="fa fa-check"></i><b>3.7</b> Data Normalization</a></li>
<li class="chapter" data-level="3.8" data-path="data-processing.html"><a href="data-processing.html#data-scaling"><i class="fa fa-check"></i><b>3.8</b> Data Scaling</a></li>
<li class="chapter" data-level="3.9" data-path="data-processing.html"><a href="data-processing.html#data-trimming"><i class="fa fa-check"></i><b>3.9</b> Data Trimming</a></li>
<li class="chapter" data-level="3.10" data-path="data-processing.html"><a href="data-processing.html#data-filtering"><i class="fa fa-check"></i><b>3.10</b> Data Filtering</a></li>
<li class="chapter" data-level="3.11" data-path="data-processing.html"><a href="data-processing.html#systematic-information"><i class="fa fa-check"></i><b>3.11</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial for R MicrobiomeAnalysis package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Tutorial for R MicrobiomeAnalysis package</h1>
<p class="author"><em>Hua Zou</em></p>
<p class="date"><em>2023-01-12</em></p>
</div>
<div id="introduction" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Introduction<a href="index.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This tutorial describes how to perform bioinformatics data analysis of microbiota using metagenomic data. It will focus on data processing, diversity analysis, biomarker identification and association investigation from microbiota expression profile which derived from <strong>amplicon (16s rRNA)</strong> or <strong>Whole genome (metagenomics)</strong> sequencing data.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-1"></span>
<img src="figures/Schematic.png" alt="Flowchat of MicrobiomeAnalysis" width="100%" height="100%" />
<p class="caption">
Figure 1.1: Flowchat of MicrobiomeAnalysis
</p>
</div>
<p>The data pre-processing module includes six main procedures:</p>
<ul>
<li><strong>Data Transformation</strong></li>
<li><strong>Data Imputation</strong></li>
<li><strong>Data Normalization</strong></li>
<li><strong>Data Scaling</strong></li>
<li><strong>Data Trimming</strong></li>
<li><strong>Data Filtering</strong></li>
</ul>
<p>The downstream analysis includes several modules:</p>
<ul>
<li><strong>Diversity analysis</strong></li>
<li><strong>Ordination analysis</strong></li>
<li><strong>Clustering analysis</strong></li>
<li><strong>Differential analysis</strong></li>
<li><strong>Association analysis</strong></li>
</ul>
<div id="input-data" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Input data<a href="index.html#input-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>when using data pre-processing functions, it requires that the inputs which are from 16s or metagenomics data should be converted into phyloseq-class object:</p>
<ul>
<li><p><strong>Types of data</strong>. Users can handle 16s or metagenomics data coming from any platform on a transformed phyloseq-class object when they use <code>import_dada2</code> or <code>import_qiime2</code>.</p></li>
<li><p><strong>Transformation</strong>. The package provides multiple methods to transform individual values, such as log10,</p></li>
<li><p><strong>Imputation</strong>. Missing value imputation methods also were provided by this package.</p></li>
<li><p><strong>Normalisation</strong>. <code>MicrobiomeAnalysis</code> also has so many methods to normalize data sample by sample, which could remove technical effects.</p></li>
<li><p><strong>Prefiltering</strong>. Removing low abundant taxa also could be done by using <code>MicrobiomeAnalysis</code>.</p></li>
<li><p><strong>Data format</strong>. Most functions of <code>MicrobiomeAnalysis</code> are using phyloseq-class object as input. Therefore, users could convert their inputs into phyloseq-class object.</p></li>
</ul>
</div>
<div id="methods" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Methods<a href="index.html#methods" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="some-background-knowledge" class="section level3 hasAnchor" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Some background knowledge<a href="index.html#some-background-knowledge" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We list here the main methodological or theoretical concepts you need to know to be able to efficiently apply <code>MicrobiomeAnalysis</code>:</p>
<ul>
<li><strong>phyloseq-class, SummarizedExperiment-class</strong>: the former object contains otu table, sample data and taxa table etc which are usually used in metagenomic data analysis. The latter one comprises expression data and metadata of gene, metabolites etc.</li>
</ul>
</div>
<div id="key-publications" class="section level3 hasAnchor" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Key publications<a href="index.html#key-publications" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p><strong>phyloseq object</strong>. <span class="citation">(<a href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and Holmes 2013</a>)</span> phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data.</p></li>
<li><p><strong>Permutational Multivariate Analysis of Variance (PERMANOVA)</strong>. <span class="citation">(<a href="#ref-anderson2014permutational" role="doc-biblioref">Anderson 2014</a>)</span> Permutational multivariate analysis of variance (PERMANOVA).</p></li>
<li><p><strong>ALDEx2</strong>. <span class="citation">(<a href="#ref-fernandes2014unifying" role="doc-biblioref">Fernandes et al. 2014</a>)</span> Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis.</p></li>
</ul>
</div>
</div>
<div id="outline-of-this-tutorial" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Outline of this tutorial<a href="index.html#outline-of-this-tutorial" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p><strong>Chapter <a href="https://zouhua.top/MicrobiomeAnalysis_book/how-to-install-microbiomeanalysis.html">2</a></strong> How to install <code>MicrobiomeAnalysis</code></p></li>
<li><p><strong>Chapter <a href="">3</a></strong> Data Processing</p></li>
</ul>
</div>
<div id="citation" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Citation<a href="index.html#citation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Kindly cite by using citation(“MicrobiomeAnalysis”) if you think <code>MicrobiomeAnalysis</code> helps you. Alternative way is Zou H (2022). MicrobiomeAnalysis: An R package for analysis and visualization in metagenomics. R package version 1.0.3, &lt;URL:<a href="https://github.com/HuaZou/MicrobiomeAnalysis/" class="uri">https://github.com/HuaZou/MicrobiomeAnalysis/</a>&gt;.</p>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-anderson2014permutational" class="csl-entry">
Anderson, Marti J. 2014. <span>“Permutational Multivariate Analysis of Variance (PERMANOVA).”</span> <em>Wiley Statsref: Statistics Reference Online</em>, 1–15.
</div>
<div id="ref-fernandes2014unifying" class="csl-entry">
Fernandes, Andrew D, Jennifer Ns Reid, Jean M Macklaim, Thomas A McMurrough, David R Edgell, and Gregory B Gloor. 2014. <span>“Unifying the Analysis of High-Throughput Sequencing Datasets: Characterizing RNA-Seq, 16S rRNA Gene Sequencing and Selective Growth Experiments by Compositional Data Analysis.”</span> <em>Microbiome</em> 2 (1): 1–13.
</div>
<div id="ref-mcmurdie2013phyloseq" class="csl-entry">
McMurdie, Paul J, and Susan Holmes. 2013. <span>“Phyloseq: An r Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.”</span> <em>PloS One</em> 8 (4): e61217.
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="how-to-install-microbiomeanalysis.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MicrobiomeAnalysis.pdf", "MicrobiomeAnalysis.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
